lib/AST/ASTContext.cpp

```diff
@@ -328,7 +328,8 @@ struct ASTContext::Implementation {
   CanGenericSignature SingleGenericParameterSignature;
 
   /// The existential signature <T : P> for each P.
-  llvm::DenseMap<CanType, CanGenericSignature> ExistentialSignatures;
+  llvm::DenseMap<std::pair<CanType, const DeclContext *>, CanGenericSignature>
```

Comment:
The uniquing key should be a GenericSignature not a DeclContext *, since the only part of the DeclContext you care about is the generic signature.